pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -776,7 +778,7 @@ cdef class StringHashTable(HashTable):
                 labels[i] = na_sentinel
             else:
                 # if ignore_na is False, we also stringify NaN/None/etc.
-                v = util.get_c_string(val)
+                get_string_data_checked(val, &v, NULL)
```

Comment:
this existing usage is more pythonic than pointer-passing.  would it be costly to keep the signature unchanged (or less-changed)?

Reply:
I can add some more wrappers (or change this particular one to return `v` instead of taking a pointer of it), that's easy.

Reply:
We have reworked the proposed patch so `get_c_string` signature is retained (but the function itself was moved from `_libs/util.pxd` to `_libs/tslibs/util.pxd`).